### PR TITLE
Fix: iOS Colors

### PR DESCRIPTION
**Pull request by tuliomir**
Solves items from #318 

### Acceptance Criteria
- Fixes colors that were off on the iOS version of the app. Details below.
- Implements the [Themes feature](https://reactnavigation.org/docs/themes/) of React Navigation v6

### Demonstrations
Below are some demonstrations of the changes. Click to enlarge the images.

| Screen | Old v0.25.0<br/><sub>Android device</sub> | Android<br/><sub>Simulator</sub> | iOS<br/><sub>Simulator</sub> |
|---|---|---|---|
| Dashboard | 01 - Dashboard - v25 | 01 - Dashboard - Android | 01 - Dashboard - iOS |
| Settings | 02 - Settings - v25 | 02 - Settings - Android | 02 - Settings - iOS |
| Send | 03 - Send - v25 | 03 - Send - Android | 03 - Send - iOS |
| Receive | 04 - Receive - v25 | 04 - Receive - Android | 04 - Receive - iOS |
| Request | 05 - Request - v25 | 05 - Request - Android | 05 - Request - iOS |

#### Border Width
| Width | iPhone SE<br/>3rd gen | iPhone 13<br/>Mini | iPhone 14<br/>Pro |
|---|---|---|---|
| 1 | iPhone SE (3rd generation) - border 1 | iPhone 13 mini - border 1 | iPhone 14 Pro - border 1 |
| 1.5 | iPhone SE (3rd generation) - border 15 | iPhone 13 mini - border 15 |iPhone 14 Pro - border 15 |
| 2 | iPhone SE (3rd generation) - border 2 | iPhone 13 mini - border 2 | iPhone 14 Pro - border 2 |

#### Dashboard
Fixes the Dashboard header color
Dashboard

Fixes the Settings header and outer container color
Settings

Improves visibility on the Header border and Input borders
Header and Input borders

Improves visibility on the Placeholder text color
Placeholder color

### Security Checklist
- [X] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: src/components/HathorHeader.js [hunk 1]
```diff
@@ -78,10 +78,9 @@ const styles = StyleSheet.create({
   wrapper: {
     height: HEADER_HEIGHT,
     flexDirection: 'row',
-    alignItems: 'center',
-    borderColor: '#eee',
+    alignItems: 'flex-end',
```
Comment (tuliomir) [inline src/components/HathorHeader.js:81]: Reverts the central alignment to a bottom alignment because of the background color change.

File: src/components/SimpleInput.js [hunk 2]
```diff
@@ -69,8 +70,8 @@ const SimpleInput = (props) => {
 
 const styles = StyleSheet.create({
   inputContainer: {
-    borderBottomWidth: 1,
-    borderColor: '#eee',
+    borderBottomWidth: STYLE.borderWidth,
```
Comment (tuliomir) [inline src/components/SimpleInput.js:73]: Increased border on iOS to improve visibility

**Review by andreabadesso — APPROVED** [review]

File: src/styles/themes.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+ * @property {string} positiveBalanceColor Represents a positive feedback for the user
+ * @property {string} errorBgColor For containers with error feedbacks
+ * @property {string} errorTextColor For texts with error messages
+ * @property {string} errorTextShadow Washed down version of error texts
+ * @property {string} primary Primary color, as set on the application config file
+ * @property {string} primaryOpacity10 Primary color washed down to 10% opacity
+ * @property {string} primaryOpacity30 Primary color washed down to 30% opacity
+ */
+export const COLORS = {
+  backgroundColor: '#fff',
```
Comment (pedroferreira1) [inline src/styles/themes.js:48]: The current code has a different background color for iOS and Android. Why do you have only one now?

File: src/styles/themes.js [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+  lowContrastDetail: '#f7f7f7',
+  midContrastDetail: '#9e9e9e',
+  borderColor: '#eee',
+  textColor: '#000',
+  textColorShadow: 'rgba(0, 0, 0, 0.5)',
+  textColorShadowLight: 'rgba(0, 0, 0, 0.3)',
+  tabBarBackground: '#333',
+  positiveBalanceColor: '#0DA0A0',
+  errorBgColor: '#DE3535',
```
Comment (pedroferreira1) [inline src/styles/themes.js:57]: We used to have these colors in the constants file. Can we delete from there?

File: src/components/AmountTextInput.js [hunk 5]
```diff
@@ -64,6 +65,7 @@ class AmountTextInput extends React.Component {
         keyboardAppearance='dark'
         keyboardType='numeric'
         placeholder={placeholder}
+        placeholderTextColor={COLORS.midContrastDetail}
```
Comment (pedroferreira1) [inline src/components/AmountTextInput.js:68]: Is it the same color for both iOS and Android?

File: src/styles/themes.js [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+  primaryOpacity30: `${PRIMARY_COLOR}4D`,
+};
+
+/**
+ * @type {{borderWidth: (number), headerHeight: number}}
+ * @property {number} borderWidth Defines border visibility throughout the app
+ * @property {number} headerHeight Defines the height of the screen title, for calculation purposes
+ */
+export const STYLE = {
+  borderWidth: Platform.OS === 'ios' ? 1.5 : 1, // 1 is barely visible on iOS
```
Comment (pedroferreira1) [inline src/styles/themes.js:71]: Before the upgrade dependencies the width was 1 for both iOS and Android? Why is it different now?

File: src/components/HathorHeader.js [hunk 7]
```diff
@@ -76,12 +75,11 @@ const HathorHeader = (props) => {
...
   wrapper: {
-    height: HEADER_HEIGHT,
+    height: STYLE.headerHeight,
     flexDirection: 'row',
-    alignItems: 'center',
-    borderColor: '#eee',
+    alignItems: 'flex-end',
+    borderColor: COLORS.borderColor,
     paddingHorizontal: 16,
-    backgroundColor: baseStyle.title.backgroundColor,
```
Comment (pedroferreira1) [inline src/components/HathorHeader.js:84]: Why did you remove this?

File: src/screens/Settings.js [hunk 8]
```diff
@@ -68,7 +67,6 @@ export class Settings extends React.Component {
       width: 100,
       marginTop: 16,
       marginBottom: 16,
-      backgroundColor: baseStyle.title.backgroundColor,
```
Comment (pedroferreira1) [inline src/screens/Settings.js:71]: Why did you remove this?

File: src/styles/themes.js (hunk 3 above)
Comment (tuliomir) [reply src/styles/themes.js:48]: Previously there was an effort to match the background color with the header color and the status bar on top. For reference, the android color was `#fafafa`, hardly noticeable from `#ffffff`.

Since the removal of the background color on the Header ( see "Dashboard" and "Settings" screenshots on the PR description, and the `HathorHeader.js` line 84 removing the background color ), there is no need to try and match with such precision the background color with the OS because of this outer border that contains the "lowContrast" color.

Having the same background color also makes the app presentation more consistent between operating systems.

What do you think about this?

File: src/styles/themes.js (hunk 4 above)
Comment (tuliomir) [reply src/styles/themes.js:57]: They are prepared to be deleted on the colors refactor on #327

File: src/components/AmountTextInput.js (hunk 5 above)
Comment (tuliomir) [reply src/components/AmountTextInput.js:68]: Yes, all colors are now the same between operating systems.

File: src/styles/themes.js (hunk 6 above)
Comment (tuliomir) [reply src/styles/themes.js:71]: Yes, before the upgrades, the width was `1` _mostly_ on all places, some were `1.5`.

The change to `1.5` on the main style is an attempt to solve item 4 of issue 318: not being able to see lines on manual inputs. With only `1` border length on the iOS simulator the lines are barely visible, indeed.

Another solution would be to keep the border at only `1` and change the `borderColor` to have a bit more contrast with the white background.

File: src/components/HathorHeader.js (hunk 7 above)
Comment (tuliomir) [reply src/components/HathorHeader.js:84]: A [discussion on slack](https://hathornetwork.slack.com/archives/G01EXLHAZUP/p1694008418391819?thread_ts=1694008353.176049&cid=G01EXLHAZUP) made me notice this background color was not consistent with what was implemented on previous versions ( 0.25.0 and before ).

This removal fixed it, as now the Header background color is the same as the rest of the main application colors. See `Dashboard` and `Settings` screenshots on the PR description for the result of this change.

File: src/screens/Settings.js (hunk 8 above)
Comment (tuliomir) [reply src/screens/Settings.js:71]: Same reason as the `HathorHeader` comment above.

File: src/styles/themes.js (hunk 6 above)
Comment (pedroferreira1) [reply src/styles/themes.js:71]: How it was before? The border color was in fact different?

File: src/styles/themes.js (hunk 3 above)
Comment (pedroferreira1) [reply src/styles/themes.js:48]: I just need to see the difference between the screens now and before. Do you have a screenshot comparing old and new screen for both iOS and Android?

File: src/styles/themes.js (hunk 6 above)
Comment (pedroferreira1) [reply src/styles/themes.js:71]: There's no before for iOS, so it's hard to compare. Seeing both iOS and Android screens, I clearly see the width difference, I feel it should be 1 for both. Can you share a picture of iOS with border 1, please?

Comment (tuliomir) [reply src/styles/themes.js:71]: Sure! Updated the PR description with borders `1`, `1.5` and `2` for the iOS.

Comment (tuliomir) [reply src/styles/themes.js]: Done on [c61c8e1](https://github.com/HathorNetwork/hathor-wallet-mobile/pull/326/commits/c61c8e1d6fb377ef605aff9f59c03ed8c9798a66) :+1:

**Review by pedroferreira1 — APPROVED** [review]
